src/category_theory/monoidal/braided.lean

```diff
@@ -283,4 +298,239 @@ def discrete.braided_functor (F : M →* N) : braided_functor (discrete M) (disc
+
+lemma tensor_μ_def₁ (X₁ X₂ Y₁ Y₂ : C) :
+    tensor_μ C (X₁, X₂) (Y₁, Y₂) ≫ (α_ X₁ Y₁ (X₂ ⊗ Y₂)).hom ≫ (𝟙 X₁ ⊗ (α_ Y₁ X₂ Y₂).inv)
+  = (α_ X₁ X₂ (Y₁ ⊗ Y₂)).hom ≫ (𝟙 X₁ ⊗ (α_ X₂ Y₁ Y₂).inv) ≫ (𝟙 X₁ ⊗ ((β_ X₂ Y₁).hom ⊗ 𝟙 Y₂)) :=
+by { dsimp [tensor_μ], simp }
+
+lemma tensor_μ_def₂ (X₁ X₂ Y₁ Y₂ : C) :
+    (𝟙 X₁ ⊗ (α_ X₂ Y₁ Y₂).hom) ≫ (α_ X₁ X₂ (Y₁ ⊗ Y₂)).inv ≫ tensor_μ C (X₁, X₂) (Y₁, Y₂)
+  = (𝟙 X₁ ⊗ ((β_ X₂ Y₁).hom ⊗ 𝟙 Y₂)) ≫ (𝟙 X₁ ⊗ (α_ Y₁ X₂ Y₂).hom) ≫ (α_ X₁ Y₁ (X₂ ⊗ Y₂)).inv :=
+by { dsimp [tensor_μ], simp }
+
+lemma tensor_μ_natural {X₁ X₂ Y₁ Y₂ U₁ U₂ V₁ V₂ : C}
+  (f₁ : X₁ ⟶ Y₁) (f₂ : X₂ ⟶ Y₂) (g₁ : U₁ ⟶ V₁) (g₂ : U₂ ⟶ V₂) :
+  ((f₁ ⊗ f₂) ⊗ (g₁ ⊗ g₂)) ≫ tensor_μ C (Y₁, Y₂) (V₁, V₂) =
+    tensor_μ C (X₁, X₂) (U₁, U₂) ≫ ((f₁ ⊗ g₁) ⊗ (f₂ ⊗ g₂)) :=
+begin
+  dsimp [tensor_μ],
+  slice_lhs 1 2 { rw [associator_naturality] },
+  slice_lhs 2 3 { rw [←tensor_comp,
+                      comp_id f₁, ←id_comp f₁,
+                      associator_inv_naturality,
+                      tensor_comp] },
+  slice_lhs 3 4 { rw [←tensor_comp, ←tensor_comp,
+                      comp_id f₁, ←id_comp f₁,
+                      comp_id g₂, ←id_comp g₂,
+                      braiding_naturality,
+                      tensor_comp, tensor_comp] },
+  slice_lhs 4 5 { rw [←tensor_comp,
+                      comp_id f₁, ←id_comp f₁,
+                      associator_naturality,
+                      tensor_comp] },
+  slice_lhs 5 6 { rw [associator_inv_naturality] },
+  simp only [assoc],
+end
+
+lemma tensor_left_unitality (X₁ X₂ : C) :
+    (λ_ (X₁ ⊗ X₂)).hom
+  = ((λ_ (𝟙_ C)).inv ⊗ 𝟙 (X₁ ⊗ X₂)) ≫
+    tensor_μ C (𝟙_ C, 𝟙_ C) (X₁, X₂) ≫
+    ((λ_ X₁).hom ⊗ (λ_ X₂).hom) :=
+begin
+  dsimp [tensor_μ],
+  have :
+      ((λ_ (𝟙_ C)).inv ⊗ 𝟙 (X₁ ⊗ X₂)) ≫
+      (α_ (𝟙_ C) (𝟙_ C) (X₁ ⊗ X₂)).hom ≫
+      (𝟙 (𝟙_ C) ⊗ (α_ (𝟙_ C) X₁ X₂).inv)
+    = 𝟙 (𝟙_ C) ⊗ ((λ_ X₁).inv ⊗ 𝟙 X₂) := by pure_coherence,
+  slice_rhs 1 3 { rw this }, clear this,
+  slice_rhs 1 2 { rw [←tensor_comp, ←tensor_comp,
+                      comp_id, comp_id,
+                      left_unitor_inv_braiding] },
+  simp only [assoc],
+  coherence,
+end
+
+lemma tensor_right_unitality (X₁ X₂ : C) :
+    (ρ_ (X₁ ⊗ X₂)).hom
+  = (𝟙 (X₁ ⊗ X₂) ⊗ (λ_ (𝟙_ C)).inv) ≫
+    tensor_μ C (X₁, X₂) (𝟙_ C, 𝟙_ C) ≫
+    ((ρ_ X₁).hom ⊗ (ρ_ X₂).hom) :=
+begin
+  dsimp [tensor_μ],
+  have :
+      (𝟙 (X₁ ⊗ X₂) ⊗ (λ_ (𝟙_ C)).inv) ≫
+      (α_ X₁ X₂ (𝟙_ C ⊗ 𝟙_ C)).hom ≫
+      (𝟙 X₁ ⊗ (α_ X₂ (𝟙_ C) (𝟙_ C)).inv)
+    = (α_ X₁ X₂ (𝟙_ C)).hom ≫
+      (𝟙 X₁ ⊗ ((ρ_ X₂).inv ⊗ 𝟙 (𝟙_ C))) := by pure_coherence,
+  slice_rhs 1 3 { rw this }, clear this,
+  slice_rhs 2 3 { rw [←tensor_comp, ←tensor_comp,
+                      comp_id, comp_id,
+                      right_unitor_inv_braiding] },
+  simp only [assoc],
+  coherence,
+end
+
+/-
+Diagram B6 from Proposition 1 of André Joyal and Ross Street,
+"Braided monoidal categories", Macquarie Math Reports 860081 (1986).
```

Comment:
Can you add this to `references.bib` and add a cite in the style used elsewhere? I think doc-gen actually hyperlinks these for you.